src/components/users/Login.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react'
+// import Axios from 'axios'
+import { debounce } from 'throttle-debounce'
+import emailValidator from 'email-validator'
+// material
+import TextField from 'material-ui/TextField'
+import RaisedButton from 'material-ui/RaisedButton'
+// custom
+// import { URL_USERS } from '../../constants/Constants'
+
+class UserRegForm extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      loginButtonDisabled: true
+    }
+  }
+
+  validate = () => {
+    this.setState({
+      loginButtonDisabled: !(this.emailIsValid() & this.passIsValid())
+    })
+  }
+
+  emailIsValid = () => {
+    const valid = emailValidator.validate(this.state.email)
+    if (valid) {
+      this.setState({
+        validEmail: ''
+      })
+    } else {
+      this.setState({
+        validEmail: 'Wrong email'
+      })
+    }
+    return valid
+  }
+  passIsValid = () => {
+    const valid = this.state.password && this.state.password.length > 5
+    if (valid) {
+      this.setState({
+        validPass: ''
+      })
+    } else {
+      if (this.state.password) {
+        this.setState({
+          validPass: 'Pass must be at least 6 simbols'
+        })
+      } else {
+        this.setState({
+          validPass: 'Enter password'
+        })
+      }
+    }
+    return valid
+  }
+
+  handleChange = event => {
+    const target = event.currentTarget
+    this.setState({
+      [target.name]: target.value
+    })
+
+    debounce(500, () => {
```

Comment:
Sorry for not telling you this earlier:
instead of using debounce in `setState()` method call provide a callback as a second param. That callback will be executed when a state will be applied. You can read more about it here: https://reactjs.org/docs/react-component.html#setstate